ethcore/src/client/client.rs

```diff
@@ -1223,15 +1224,19 @@ impl Client {
 	// transaction for calling contracts from services like engine.
 	// from the null sender, with 50M gas.
 	fn contract_call_tx(&self, block_id: BlockId, address: Address, data: Bytes) -> SignedTransaction {
-		let from = Address::zero();
+		self.contract_call_tx_from_sender(block_id, address, Address::zero(), data)
```

Comment:
Since it's repeated so many times it should be just default implementation, but instead best to have a look at the `CallOptions` pattern above.